Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift

```diff
@@ -0,0 +1,36 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+/// A ``ValueRepository`` that allows to store any ``KnowledgeSource``s.
+public typealias UniversalValueRepository = ValueRepository<Any>
+
+
+/// A ``SharedRepository`` implementation that itself is a value type.
+///
+/// - Note: The ``ValueRepository`` is considered `Sendable` if the ``RepositoryAnchor`` conforms to `Sendable`.
+public struct ValueRepository<Anchor>: SharedRepository, BuiltinRepository {
+    var storage: Container = [:]
+
+    public init() {}
+
+    public func get<Source: KnowledgeSource<Anchor>>(_ source: Source.Type) -> Source.Value? {
+        get0(source)
+    }
+
+    public mutating func set<Source: KnowledgeSource<Anchor>>(_ source: Source.Type, value newValue: Source.Value?) {
+        set0(source, value: newValue)
+    }
+
+    public func collect<Value>(allOf type: Value.Type) -> [Value] {
+        collect0(allOf: type)
+    }
+}
+
+
+extension ValueRepository: @unchecked Sendable where Anchor: Sendable {}
```

Comment:
Should we somehow add a locking mechanism to enforce some Sendable mechanisms here? 
I might miss something, I would say that by being a struct we should not really have a problem here and it is more up to `Container` to enforce anything here and it is not modifiable anyways. Am I on the correct path? Might be good to add a comment above the `@unchecked` conformance to ensure that we document the reasoning here if we don't use a locking mechanism.

Reply:
I tried to somehow add a runtime check but failed with all my attempts. It's really sad that Swift won't allow you to check for `is Sendable`.